pkg/cli/service.go

```diff
@@ -0,0 +1,302 @@
+
+	table := getNoBorderTableWriter(&builder)
+
+	for _, tag := range img.Tags {
+		imageName := ellipsize(img.Name, imageNameWidth-padding, ellipsis)
+		tagName := ellipsize(tag.Name, tagWidth-padding, ellipsis)
+		digest := ellipsize(tag.Digest, digestWidth-padding, "")
+		size := ellipsize(strings.ReplaceAll(humanize.Bytes(tag.Size), " ", ""), sizeWidth-padding, ellipsis)
+		row := []string{imageName,
+			tagName,
+			digest,
+			size,
+		}
+
+		table.Append(row)
+	}
+
+	table.Render()
+
+	return builder.String(), nil
+}
+
+func (img imageStruct) stringJSON() (string, error) {
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
+	body, err := json.MarshalIndent(img, "", "  ")
+
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func (img imageStruct) stringYAML() (string, error) {
+	body, err := yaml.Marshal(&img)
+
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+type catalogResponse struct {
+	Repositories []string `json:"repositories"`
+}
+
+type manifestResponse struct {
+	Layers []struct {
+		MediaType string `json:"mediaType"`
+		Digest    string `json:"digest"`
+		Size      uint64 `json:"size"`
+	} `json:"layers"`
+	Annotations struct {
+		WsTychoStackerStackerYaml string `json:"ws.tycho.stacker.stacker_yaml"`
+		WsTychoStackerGitVersion  string `json:"ws.tycho.stacker.git_version"`
+	} `json:"annotations"`
+	Config struct {
+		Size      int    `json:"size"`
+		Digest    string `json:"digest"`
+		MediaType string `json:"mediaType"`
+	} `json:"config"`
+	SchemaVersion int `json:"schemaVersion"`
+}
+
+func combineServerAndEndpointURL(serverURL, endPoint string) (string, error) {
+	if !isURL(serverURL) {
+		return "", zotErrors.ErrInvalidURL
+	}
+
+	newURL, err := url.Parse(serverURL)
+
+	if err != nil {
+		return "", zotErrors.ErrInvalidURL
+	}
+
+	newURL, _ = newURL.Parse(endPoint)
+
+	return newURL.String(), nil
+}
+
+func ellipsize(text string, max int, trailing string) string {
+	if len(text) <= max {
+		return text
+	}
+
+	chopLength := len(trailing)
+
+	return text[:max-chopLength] + trailing
+}
+
+func getNoBorderTableWriter(writer io.Writer) *tablewriter.Table {
+	table := tablewriter.NewWriter(writer)
+
+	table.SetAutoWrapText(false)
+	table.SetAutoFormatHeaders(true)
+	table.SetHeaderAlignment(tablewriter.ALIGN_LEFT)
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetCenterSeparator("")
+	table.SetColumnSeparator("")
+	table.SetRowSeparator("")
+	table.SetHeaderLine(false)
+	table.SetBorder(false)
+	table.SetTablePadding("   ")
+	table.SetNoWhiteSpace(true)
+	table.SetColMinWidth(colImageNameIndex, imageNameWidth-padding)
+	table.SetColMinWidth(colTagIndex, tagWidth-padding)
+	table.SetColMinWidth(colDigestIndex, digestWidth-padding)
+	table.SetColMinWidth(colSizeIndex, sizeWidth-padding)
+
+	return table
+}
+
+const (
+	imageNameWidth = 32
+	tagWidth       = 16
+	digestWidth    = 8
+	sizeWidth      = 8
+	padding        = 0
+	ellipsis       = ".."
```

Comment:
... three dots better

Reply:
and elipsis handling for all columns?

Reply:
Yes. All of them are being shortened if necessary using ellipsize().. I am calling that for all fields